src/cache/cache.jl

```diff
@@ -142,6 +146,14 @@ function build_cache(Y, atmos, params, surface_setup, sim_info)
     net_energy_flux_toa = [Geometry.WVector(FT(0))]
     net_energy_flux_sfc = [Geometry.WVector(FT(0))]
 
+    conservation_check =
+        atmos.surface_model isa PrognosticSurfaceTemperature ?
+        (;
+            col_integrated_precip_energy_tendency = zeros(
+                axes(Fields.level(Geometry.WVector.(Y.f.u₃), half)),
+            )
+        ) : (; col_integrated_precip_energy_tendency = (;))
```

Comment:
It might be good to always have this cache. We will need it for atmosphere only energy check anyway.

Reply:
But wouldn't this add unnecessary variables to dry runs? Or do we not want to run ClimaAtmos dry anymore? 🤔

Reply:
Ah, right. I meant having this whenever precip is not `nothing`. (i.e. not dependent on `PrognosticSurfaceTemperature`)

Reply:
@szy21, it seems that by generalizing this as suggested above, we include some jobs that are also checked by flame graphs. These checks fail because this check adds substantially more allocations (more than expected- see [build](https://buildkite.com/clima/climaatmos-ci/builds/16296)). Would you prefer that we 1) only make this applicable to the slab ocean example for now and try to deal with the allocation issue as part of the next PR; or 2) increase the allowable allocation limit?

Reply:
I think this [line](https://github.com/CliMA/ClimaAtmos.jl/blob/a079b1ddfeea16c7e3f349adc70f4673528d0955/src/parameterized_tendencies/microphysics/precipitation.jl#L114) is allocating. Could you try caching `col_precip_energy` as what is done for e.g. `col_integrated_rain` and see if it helps with allocation? It seems you can use `col_precip_energy` directly instead of `col_integrated_precip_energy_tendency` in the surface temperature equation, and that shouldn't allocate.

Reply:
This normally shouldn't allocate. I tried it anyway, and it doesn't make that much difference. :-(

Reply:
see [this build](https://buildkite.com/clima/climaatmos-ci/builds/16299#018d58e1-ff70-4885-923c-8dca1a99f563)

Reply:
It's probably to do with [this](https://github.com/CliMA/ClimaAtmos.jl/blob/b4ef211c25426e71233912cc493af8293ec81b0c/src/parameterized_tendencies/microphysics/precipitation.jl#L162C1-L169C14).

Reply:
Oh! It might be you need to have a temporary field for
```
@. ᶜS_ρq_tot[colidx] * e_tot_0M_precipitation_sources_helper(
                    thermo_params,
                    ᶜts[colidx],
                    ᶜΦ[colidx],
                )
```
Otherwise it may allocate a new field for the integral? I can give it a try.

Reply:
This works locally: https://github.com/CliMA/ClimaAtmos.jl/blob/f5559a40a3cc1189fc03160816073adb77ddbdb7/src/parameterized_tendencies/microphysics/precipitation.jl#L155-L167

Reply:
yep, [here's](https://buildkite.com/clima/climaatmos-ci/builds/16300) the test.

Reply:
Yeah, just use the temporary field as above should fix it.

Reply:
(You can probably use ᶜtemp_scalar instead of ᶜtemp_scalar_3. I didn't check the entire file when trying it.)

Reply:
Ahh thank you. Sorry, GH didn't refresh, so I hadn't seen your suggestion above. Just running it now.

Reply:
🎉